.github/workflows/npm-release.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish on NPM
+
+on:
+  workflow_dispatch:
+    inputs:
+      version_type:
+        description: 'Version to release, in valid semver format'
+        required: true
+        type: string
+
+jobs:
+  publish-to-npm:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    environment:
+      name: npm
+      url: https://www.npmjs.com/package/@morpho-org/morpho-blue
+    steps:
+
+      - name: Validate semver format of the provided version
+        run: |
+          if [[ ! "${{ github.event.inputs.version_type }}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+              echo "Invalid semver format. Please provide a valid semver format (v1.2.3)"
+              exit 1
+          fi
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Bump package.json version
+        run: |
+          version_type=$(echo "${{ steps.determine_version.outputs.version_type }}")
+          yarn version --new-version ${version_type} --no-git-tag-version
```

Comment:
The `--new-version` parameter seems to be available with `yarn publish`: https://classic.yarnpkg.com/lang/en/docs/cli/publish/#toc-yarn-publish-new-version

Reply:
Just tested it: it doesn't commit it to git
But it's actually a benefit because it doesn't break the commit msg syntax convention that we use (whereas `yarn version` commits `vx.x.x` as commit)
So it means using `yarn publish --new-version` allows to publish the new version and optionally commit it with a defined commit msg

Also, committing the version **after** having published it seems better because in case the `publish` step fails, no git tag/commit is pushed

Reply:
I am surprised they recommend to push the commit right after versioning: https://classic.yarnpkg.com/lang/en/docs/cli/version/#toc-version-lifecycle

But I guess it only depends on what we want. So no strict requirement on this from my end in the end

I also discover: `yarn config set version-git-message` which I think is important though, for our commit msg convention

Reply:
Isn't there a way to tell `yarn` to not version anything and just use the package's version defined in `package.json`?

So the flow would be:
1. commit a new version to `package.json`
2. merge
3. trigger action without any input
4. if action is triggered twice with the same version, fail

Reply:
That's a good tradeoff. Ive also reduced the version to 0.1.0, and we'll release the 1.0.0 once deployed (it should never change after)

Reply:
This discussion can be resolved but I keep it open for others to know